src/EnergyPlus/HybridEvapCoolingModel.cc

```diff
@@ -0,0 +1,2389 @@
+			//Supply air mass flow rate ratio describes supply air mass flow rate as a fraction of mass flow rate associated with the value in field : "System Maximum Supply Air Flow Rate".
+			//If this field is blank, the lower constraint on outside air fraction will be 0.00,default 0.10
+			Min_Msa = minMsa;
+			Max_Msa = maxMsa;
+			return true;
+		}
+		bool CMode::ValidPointer(int curve_pointer) {
+			if (curve_pointer >= 0) return true;
+			else  return false;
+		}
+		Real64 CMode::CalculateCurveVal( Real64 X_1, Real64 X_2, Real64 X_3, Real64 X_4, Real64 X_5, Real64 X_6, int curveType)
+		{
+		// SUBROUTINE INFORMATION:
+		//       AUTHOR         Spencer Maxwell Dutton
+		//       DATE WRITTEN   October 2017 
+		//       MODIFIED       
+		//       RE-ENGINEERED  na
+
+		// PURPOSE OF THIS SUBROUTINE:
+		// Returns the normalized or scaled output from the performance curve as specified by curveType.  6 independent variables are required. 
+
+		// METHODOLOGY EMPLOYED:
+		// Makes a call to the curve manager, then multiplies the result by either the NormalizationReference for intensive variables or by the scaling Correction
+		// for extensive variables.
+		// The Normalization reference is specified as a model input, the Correction is equal to the ScaledSystemMaximumSupplyAirMassFlowRate, which is calculated 
+		// as  Correction=ScaledSystemMaximumSupplyAirMassFlowRate= SystemMaximumSupplyAirFlowRate*ScalingFactor*AirDensity, where SystemMaximumSupplyAirFlowRate, and ScalingFactor are 
+		// model inputs.
+		// The following tables are for intensive variables : Supply Air Temperature, Supply Air Humidity, External Static Pressure,
+		// The following tables are for extensive variables : Electric Power, Fan Electric Power, Second Fuel Consumption, Third Fuel Consumption, Water Use Lookup Table
+		// 
+		// X_1 is the outside air temperature (Tosa), X_2 is the outside humidity ratio (Wosa), 
+		// X_3 return air temp (Tra),X_4 return humidity ratio Wra,
+		// X_5 supply air mass flow rate, Msa, X_6 outside air fraction OSAF
+		// the curveType 
+
+		// REFERENCES:
+		// na
+
+		// Using/Aliasing
+			Real64 Y_val = 0;
+
+			switch (curveType)
+			{
+			case TEMP_CURVE:
+				if (ValidPointer(Tsa_curve_pointer))
+				{
+					Y_val = NormalizationReference*CurveValue(Tsa_curve_pointer, X_1, X_2, X_3, X_4, X_5, X_6);
+				}
+				else {
+					Y_val = X_3;//return air temp
+				}
+				break;
+			case W_CURVE:
+
+				if (ValidPointer(HRsa_curve_pointer))
+				{
+					Y_val = NormalizationReference*CurveValue(HRsa_curve_pointer, X_1, X_2, X_3, X_4, X_5, X_6);//0.01;//
+				}
+				else {
+					Y_val = X_4;//return HR
+				}
+				break;
+			case POWER_CURVE:
+
+				if (ValidPointer(Psa_curve_pointer))
+				{ // Correction= scaling factor *System Maximum Supply Air Flow Rate*StdRhoAir
+					Y_val = Correction*CurveValue(Psa_curve_pointer, X_1, X_2, X_3, X_4, X_5, X_6);
+				}
+				else {
+					Y_val = 0;//or set a more reasonable default
```

Comment:
This and the below items say "or set a more reasonable default".  So...should you be setting a more reasonable default?  If not, shouldn't the curve just be required?

Reply:
Yes thats a good point, I will chat with Jonathan about what we really want to happen if a user doesn't give us a valid curve for power use. Should we just give a warning and return 0 for power in that mode?  Maybe.